### PR TITLE
Article jumping fixes and refactoring

### PR DESCRIPTION
**Pull request by vedgy**
See the commit messages for details.

### CODE REVIEW

**Comment by xiaoyifang** [comment]
what about windows size changes, do the variables  sx,sy  need to be change?

**Comment by vedgy** [comment]
> what about windows size changes, do the variables sx,sy need to be change?

What do you mean? Which commit/change should require additional changes to sx and sy?

File: articleview.cc [hunk 1]
```diff
@@ -566,54 +566,51 @@ void ArticleView::loadFinished( bool )
...
-        ui.definition->page()->mainFrame()->evaluateJavaScript(
-            QString( "window.scroll( %1, %2 );" ).arg( sx ).arg( sy ) );
-      }
+    const QString currentArticle = userData.value( "currentArticle" ).toString();
+    if( !currentArticle.isEmpty() )
+    {
+      // There's a current article saved, so set it to be current.
+      // If a scroll position was stored - even (0, 0) - don't move to the
+      // current article.
+      setCurrentArticle( currentArticle, moveToCurrentArticle );
```
Comment (vedgy) [inline articleview.cc:603]: Reordering `window.scroll` and `setCurrentArticle()` was probably a bad idea. What if a possibly user-customized current article style changes its height? I am going to revert the order to what it was before this change.

Comment (vedgy) [reply articleview.cc:595]: Done.

**Comment by xiaoyifang** [comment]
the variables `sx,sy` are used to scroll to the position when page was navigated back through history .
can not the WebKit engine remember the page's position?

**Comment by vedgy** [comment]
> Can't the WebKit engine remember the history page's position?

See the 7d3537243d088520a3d93182cc762c88166503d2 commit message. I don't know whether saving&restoring these coordinates will be needed for websites with Qt WebEngine. My initial port won't support external websites properly anyway.

**Pull request merged**